Add random query to userstyle link href

Tacodiva opened the pull request:
Possibly resolves #5787

apple502j reviewed (approved):
Maybe timestamps? though this looks okay

WorldLanguages reviewed (changes requested):
Unfortunately some other parts of the cs.js file don't expect the href property to be changed this way. Those will need to be changed from `.href ` to `.getAttribute("data-style-href")`.
We'd also have to double check for side effects in other files. There's 1 or 2 addons that directly interact with their <link> elements, hopefully not through their href property.

WorldLanguages commented:
I just committed some progress, but it remains untested. And I'm not sure if I missed some other file (or line within cs.js) that needs to be changed too.

Samq64 reviewed (changes requested):
> And I'm not sure if I missed some other file (or line within cs.js) that needs to be changed too.

Things are still broken in a lot of places.
![image](https://user-images.githubusercontent.com/81489795/228614129-7afec601-81c7-4e9a-8c01-2e4b5ad1f574.png)

WorldLanguages commented:
This line of code won't be a problem because the userscript manually adds the link element to the DOM (not the extension), but we need to be careful if any other addons are expecting URLs to end in `.css`, since that wouldn't be the case anymore because we would be adding URL params:

https://github.com/ScratchAddons/ScratchAddons/blob/61fd2a948df8cf4cc97a2265491e21c64e068c35/addons/dark-www/experimental_forums_preview.js#L20

Tacodiva commented:
Just looked into this again quickly. All the issues where just being caused by a mistake w_l made which I have fixed. 
I also did a dirty search in the repo for js files containing `.css` and I'm pretty sure "experimental_forums_preview" is the only problematic file, which I have fixed.

I've done some testing and it seems to me like this is good to go.

Tacodiva commented:
Ping! There has been no activity for 28 days.

((I think this should be a part of the 1.35 milestone given it has potential (and has actually) caused some major and hard to diagnose issues))
